include/deal.II/base/synchronous_iterator.h

```diff
@@ -88,6 +88,34 @@ SynchronousIterators (const SynchronousIterators &i)
 
 
 /**
+ * Return a const reference to the tuple of iterators stored in the
+ * synchronous iterator.
+ */
+template <typename Iterators>
+inline
+const Iterators &
+operator* (const SynchronousIterators<Iterators> &a)
```

Comment:
I think it might actually be nicer to make this a member function of the class. Could you try whether that works? This may also allow (thinking about) making the `iterators` variable a private member. (Not necessary to do the latter as part of the former.)